### ISSUE TITLE
CI: daemon/cmd test panic'ed in TravisCI

### ISSUE THREAD

**Issue by giorio94** [cilium/cilium#25235]
## CI failure

Hit in https://github.com/cilium/cilium/pull/25221: ~~https://app.travis-ci.com/github/cilium/cilium/builds/262730618#L637-L657~~ (apparently the logs have been overwritten after restarting the job).

```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━���━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓                                                                                                  
┃   PANIC  package: github.com/cilium/cilium/daemon/cmd • Test   ┃                                                                                                  
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛                                                                                                  

panic: send on closed channel                                                                                                                                       

goroutine 787 [running]:                                                                                                                                            
github.com/cilium/cilium/pkg/identity/cache.(*localIdentityCache).lookupOrCreate(0x4001e2a3c0, 0x4000928930, 0x0?)                                                  
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/identity/cache/local.go:109 +0x438                                                                             
github.com/cilium/cilium/pkg/identity/cache.(*CachingIdentityAllocator).AllocateIdentity(0x4001a4b2c0, {0x3425ef8, 0x4000252500}, 0x4000928930, 0x80?, 0x40?)       
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/identity/cache/allocator.go:339 +0x55c                                                                         
github.com/cilium/cilium/pkg/ipcache.(*IPCache).resolveIdentity(0x40005bfe00, {0x3425ef8, 0x4000252500}, {{{0x400254efc0?, 0x0?}, 0x4000114030?}, 0x0?}, 0x0?, 0x0?)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/ipcache/metadata.go:436 +0xb24                                                                                 
github.com/cilium/cilium/pkg/ipcache.(*IPCache).InjectLabels(0x40005bfe00, {0x3425ef8, 0x4000252500}, {0x400092e9c0, 0x3, 0x3})                                     
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/ipcache/metadata.go:197 +0x4a8                                                                                 
github.com/cilium/cilium/pkg/ipcache.(*IPCache).TriggerLabelInjection.func1({0x3425ef8, 0x4000252500})                                                              
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/ipcache/metadata.go:547 +0x50                                                                                  
github.com/cilium/cilium/pkg/controller.(*Controller).runController(0x4000bbe120)                                                                                   
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/controller/controller.go:216 +0x1f0                                                                           
created by github.com/cilium/cilium/pkg/controller.(*Manager).updateController                                                                                      
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/controller/manager.go:111 +0x8e8                                                                               
FAIL	github.com/cilium/cilium/daemon/cmd	12.711s
```

**Comment by tommyp1ckles** [comment]
Wrong link? Don't see any panics in the highlighted section?

**Comment by giorio94** [comment]
> Wrong link? Don't see any panics in the highlighted section?

I did restart the job, and I think that it has overwritten the previous log messages :-(. It was the exact same message that is reported in the issue description though.

**Comment by tommyp1ckles** [comment]
Think im seeing this happen when I run locally too, I'll put up a fix for this.

**Comment by bimmlerd** [comment]
I think I hit this again, in a branch containing 3907a9a11f1c6f39b1e64794416a30d36e284a5f:

https://github.com/cilium/cilium/pull/25197, https://github.com/cilium/cilium/actions/runs/4927642135/jobs/8804911353?pr=25197#step:8:233

**Comment by bimmlerd** [comment]
I think @tommyp1ckles fixed a different, but related issue. I managed to reproduce this reliably by increasing the likelyhood of the race by inserting a sleep just before the write to the channel in [lookupOrCreate](https://github.com/cilium/cilium/blob/6f6296ffc18c0e236f9be4cf636af93b638b773e/pkg/identity/cache/local.go#L109). With the proposed fix in #25353, I didn't manage to repro anymore.
